File: src/main/java/org/featx/spec/exception/BusinessException.java
package org.featx.spec.exception;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Excepts
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class BusinessException extends RuntimeException {

    private BusinessError businessError;

    private Object extra;

    @Override
    public String getMessage() {
        return businessError.getMessage();
    }
}

File: src/main/kotlin/org/featx/jvmon/exception/BusinessException.kt
package org.featx.jvmon.exception

import lombok.Getter

/**
 * @author Excepts
 */
class BusinessException private constructor(@field:Getter val businessError: BusinessError,
                                            @field:Getter val extra: Any?) : RuntimeException() {

    override val message: String?
        get() = businessError.message

    companion object {
        fun of(businessError: BusinessError, vararg extra: Any?): BusinessException {
            return BusinessException(businessError, extra)
        }
    }
}